flink-runtime/src/main/java/org/apache/flink/runtime/checkpoint/filemerging/FileMergingSnapshotManagerBase.java

```diff
@@ -289,6 +308,13 @@ protected Path generatePhysicalFilePath(Path dirPath) {
         return new Path(dirPath, fileName);
     }
 
+    @VisibleForTesting
+    boolean isResponsibleForFile(Path filePath) {
```

Comment:
It is used only for testing for now and will also be used when integrating the recovery (FLINK-32080).
Since it accesses some private variables, I suggest keep it here.